carbonj.service/src/main/java/com/demandware/carbonj/service/engine/MetricList.java

```diff
@@ -59,22 +60,26 @@ public MetricList(MetricRegistry metricRegistry,  String name, File confFile, St
 
     public boolean match(String name)
     {
-        if ( patterns.isEmpty() )
+        if ( patternsAndCounters.isEmpty() )
         {
             return false;
         }
 
-        List<Pattern> currentPatterns = patterns; // copy so we don't keep hitting the volatile barrier
-        for ( Pattern p : currentPatterns )
+        List<Pair<Pattern, Counter>> currentPatternsAndCounters = patternsAndCounters; // copy so we don't keep hitting the volatile barrier
+        for ( int i = 0; i < currentPatternsAndCounters.size(); i++ )
         {
-            if( ".*".equals( p.pattern() ) )
+            Pair<Pattern, Counter> p = currentPatternsAndCounters.get(i);
+
+            if( ".*".equals( p.getLeft().pattern() ) )
             {
+                patternsAndCounters.get(i).getRight().inc();
                 droppedMetrics.inc();
                 return true;
             }
 
-            if ( p.matcher( name ).find() )
+            if ( p.getLeft().matcher( name ).find() )
```

Comment:
Can you also profile the run time of the pattern match and send the result to logs? It can be at debug level. This can help us identify slow running / inefficient regexes.